[1.21.1] New Growth Update

Spearkiller opened the pull request:
_NOTE: These changes are intended to go live alongside the 1.21.1 update._

The New Growth Update is here! After many years, a HUGE update to Forestry's tree related content. A bunch of new tree species, lots of changes to old ones, and a bunch of other changes inbetween. An overview of the changes can be found below.

### Additions:
- Added 14 new Tree Species.
  - Golden Elm, Balsam Fir, Coconut, Copper Beech, Feijoa, Flowering Dogwood, Ginkgo, Jacaranda, Pewen, Macrocarpa, Olive, Orange, Pear, Kauri.
- Added 5 new fruits.
  - Coconut, Pear, Orange, Feijoa, Olive.
- Added Amber, Amber Drones, Amber Saplings, Blocks of Amber, and Amber Electron Tubes
  - Amber Drones and Amber Saplings are occasionally dug up by sniffers, which can be processed into Relic drones and Ginkgo Saplings, respectively, with a 25% chance of success.
    - This replaces the previous method of obtaining Relic drones, wherein they would be dug up directly.
  - Amber is also produced when processing vintage combs.
  - Amber Electron Tubes can be used to increase the luck-based yeilds of Squeezers and Centrifuges.

### Content Changes:
- Big changes to some of the original Forestry trees.
  - Hill Cherry -> Sour Cherry
    - This was done as Hill Cherries were basically the same species as Cherry Blossoms. In addition to this, Blossom trees do not produce edible fruit.
  - Bull Pine -> Ponderosa Pine
    - This was done as 'Bull Pine' is quite a generic name, and it didn't quite match the look of what the bull pine actually was in the mod already. Ponderosa Pine was closer to what Bull Pine previously looked like, and was also a 'type' of Bull Pine, so it made sense.
  - Desert Acacia -> Camelthorn
    - This was done as the species (Vachellia erioloba) was reclassified from an Acacia some time around 2009. I was also not a huge fan of how they looked.
  - While the trees have undergone renames and reclassifications, the plank textures have remained the same, and the bark textures similar enough as to not ruin any buildings that use them.
- Changed the generation of some other Forestry trees.
  - Lemon, Plum, Pine, Spruce, Acacia, Teak, Balsa, Papaya, Padauk and Poplar.
- Resprited some log textures and colours for other trees.
- Updated sprites for all Forestry leaves, fruits, and added some new leaf blocks for existing trees.
- Resprited pretty much all Forestry saplings (again.)
- Rebalanced the existing Forestry fruits.
  - Dates, Cherries, Walnuts, and Chestnuts can be eaten much quicker.
  - Reduced the amount of seed oil produced by walnuts and chestnuts.
  - All fruits now have different hunger/saturation values.
  - Cherries now produce fruit juice instead of seed oil.
  - Papayas and Chestnuts have quite a high hunger to saturation ratio.
- Renamed some wood/sapling types.
  - Sipiri -> Greenheart
  - Yellow Meranti -> Mahogany
  - Citrus -> Lemon
  - Zebrawood -> Zebrano
  - Red Spruce -> Black Spruce
  - Ipe -> Yellow Ipe
  - Palm -> Date
- Changed nearly all of the tree mutations for each tree.
  - There is now only one mutation for each species
  - This was done to separate things into six distinct 'lines' of trees: Temperate, Swamp, Floral, Conifers, Jungle, and Arid.
  - They have also been reorganised in the creative menu/JEI to match this.
  - The last trees in each line (I.E. Chestnut, Giant Sequoia, Mahogany, etc.) are now much harder to obtain, requiring them to be bred in their preferred climate.
    - This was done to make each tree feel a bit more special, requiring a bit more effort, hopefully making them feel like a bit more of an achievement.
- Mulch and Compost can now be used as Bonemeal.
- Fertiliser (as well as Mulch and Compost) make sound when used.
- Resprited some miscellaneous things
    - Pollen, Foresters Manual, Cherry, Humus
- All trees now have descriptions when viewing them in the Analyser.
- Removed a recipe for 4x candles using beeswax
	
### Technical Changes:
- Trees now use TextColor instead of Color, fixing some weird issue that I know was an issue but I don't know enough about it.
- Added a some new methods in FeatureHelper such as generateEllipsoid() and generateLine(), as well as adding a bunch of overloaded methods for existing ones.
- Reordered the taxonomy of all trees, fixing a bunch of inaccuracies.
  - All trees' rankings are now based off of a consistent source as well (Catalogue of Life.)
- Fixed a depreciation warning in DefaultTreeSpecies.
- Changed how FeatureHelper.generateTrunk() works, as it now returns a list of BlockPos, used for keeping track of the curve of a tree caused by the lean, which was previously impossible (and very annoying!)

### Fixes:
- Fixed some inconsistent shading on Doors and Trapdoors.

Hopefully this brings some new life into the tree side of things, and encourages players to get involved in tree breeding, because I think it's fun!

Spearkiller committed:
Resprited more assets. Details in readme

Spearkiller committed:
Resprited more assets. Details in readme

Spearkiller committed:
Datagen for Elm

Spearkiller committed:
Files required to add new tree breed (Elm)

Spearkiller committed:
Trees now use TextColor instead of Color

Spearkiller committed:
Added generateEllipsoid to FeatureHelper

Spearkiller committed:
Elm Trees are now 'good enough'

Spearkiller committed:
Updated assets for Elm trees

Spearkiller committed:
updated elm logs again

Spearkiller committed:
Golden Elms are a little bit more golden

Spearkiller committed:
Update elm_door.png

Spearkiller committed:
Fixed some texture oversights

Spearkiller committed:
Fixed some texture oversights

Spearkiller committed:
updated sapling textures

Spearkiller committed:
updated poplar sapling

Spearkiller committed:
Resprited Maple Leaves

Spearkiller committed:
Recoloured Willow leaves

Spearkiller committed:
Updated palm leaves

Spearkiller committed:
Added Azalea and Ginkgo Leaf types

Spearkiller committed:
removed weird thing from jungle leaves

Spearkiller committed:
Added Ginkgo to ForestryLeafSprites

Spearkiller committed:
Create HOW TO - Add new trees.txt

Spearkiller committed:
Assets for new tree breeds

Spearkiller committed:
Updated Taxa

Conifers are now Gymnosperms.

Spearkiller committed:
Run datagen for new assets

Spearkiller committed:
Fixed directory of assets

Spearkiller committed:
Added 15 new tree species

Spearkiller committed:
Finished Balsam Fir

Spearkiller committed:
Added Coconut Tree (unfinished)

Spearkiller committed:
Made it so ellipsoids can be 'inflated' and fixes generation

Spearkiller committed:
Changed the rounding of generateEllipsoid

Spearkiller committed:
Fixed ridiculous Golden Elm generation

Spearkiller committed:
Added new cylinder and trunk generation methods

Spearkiller committed:
Basically finished Fir trees

Spearkiller committed:
Added generateLine to FeatureHelper

Spearkiller committed:
Added Coconut Trees

Spearkiller committed:
Removed commented code and unused imports

Spearkiller committed:
Added Copper Tree

Spearkiller committed:
Added new constructor to FeatureTree to allow overriding min height

Spearkiller committed:
Added Feijoa Trees

Fruits will need to be added later

Spearkiller committed:
fixed leaf offset calculations in beech and elms

Spearkiller committed:
Added all the new fruits

Spearkiller committed:
WIP textures for Jacaranda and Dogwood blossoms

Spearkiller committed:
Update FeatureDogwood.java

Spearkiller committed:
Minor changes to dogwood

Spearkiller committed:
Added Ginkgo trees

Spearkiller committed:
Added Jacaranda Trees

Spearkiller committed:
Added Monkey Puzzle Trees

Spearkiller committed:
fixed new fruit alleles not having localised names

Spearkiller committed:
Added some supporting branches to Monkey Puzzle

Spearkiller committed:
Added Macrocarpa trees

Spearkiller committed:
Added Olive Trees

Spearkiller committed:
removed debug method from olive trees (lol)

Spearkiller committed:
50% done on orange and lemon trees

Spearkiller committed:
Made vanilla cherry and acacia recessive species

Spearkiller committed:
Fixed depreciation warnings in DefaultTreeSpecies.java

Spearkiller committed:
Finished Orange Trees

Spearkiller committed:
cleaned up orange tree code

Spearkiller committed:
Updated plum, lemon and willow sapling textures

Spearkiller committed:
Updated Lemon Tree Generation

Spearkiller committed:
Reduced base height of coconut tree

Spearkiller committed:
Added Pear Trees

Spearkiller committed:
Finished Juniper?

Spearkiller committed:
updated shades of ginkgo leaves

Spearkiller committed:
Changed some tree colours + plum height

Spearkiller committed:
Remastered plum tree generation

Spearkiller committed:
Modified pear tree generation somewhat

Spearkiller committed:
Rebalanced fruits!

Spearkiller committed:
Rerun runData, hopefully fixed build fail

Spearkiller committed:
reran rundata

Spearkiller committed:
Removed Gum and Eucalyptus - They will return later

Spearkiller committed:
differentiated leaf colours

also resprited macrocarpa, monkey puzzle, and (slightly) pear saplings - and recoloured palm

Spearkiller committed:
Renamed some woods

Spearkiller committed:
Updated Olive Tree generation

Spearkiller committed:
Updated willow leaves

Spearkiller committed:
Reworked Poplar generation

Spearkiller committed:
Reclassified and renamed some trees

Spearkiller committed:
Mokala -> Camelthorn. Same tree, different name

Spearkiller committed:
Fixed a typo in taxonomy

Spearkiller committed:
Updated Balsa generation + stripped logs

Spearkiller committed:
Updated Teak logs

Spearkiller committed:
Changed teak door colouring

Spearkiller committed:
Fixed a few teak things

Spearkiller committed:
Acacia -> Camelthorn

Spearkiller committed:
Fixed inconsistent shading on a lot of wood doors

Also retextured Acacia Stripped logs, and reshaded Cocobolo. They're similar enough changes to all fall under this commit, mind.

Spearkiller committed:
Ditto, but greenheart

I forgot to save the files

Spearkiller committed:
Update stripped_hill_cherry_log.png

Spearkiller committed:
fixed some signs using the wrong log post texture

Spearkiller committed:
reshaded ipe logs

Spearkiller committed:
Reshaded Maple wood

Spearkiller committed:
Reshaded and resprited Pine

Spearkiller committed:
Retextured Mahogany wood

Spearkiller committed:
Resprited most saplings

Spearkiller committed:
Cypress -> Macrocarpa

Spearkiller committed:
Juniper -> Kauri

Spearkiller committed:
tidy-up from last commit

Spearkiller committed:
Ditto

Spearkiller committed:
Hill Cherry -> Sour Cherry, Cherry -> (Cherry) Blossom

Probably the easiest way to deal with this conflict is to just move the goalposts

Spearkiller committed:
Undone changes regarding Cherry -> Blossom

Spearkiller committed:
Updated Sour Cherry Generation

Spearkiller committed:
Changes to Black Spruce and Bull Pine generation

Bull Pine is also now Ponderosa Pine

Spearkiller committed:
Fixed improper link to CoL on Ponderosa Pine

Spearkiller committed:
Slight changes to Fir generation

Spearkiller committed:
Kauri trees are probably done?

Spearkiller committed:
Added a method to get a wood type from a log type

Spearkiller committed:
Finished Kauri

Spearkiller committed:
Updated Pollen texture

Spearkiller committed:
fixed old naming of Ponderosa Pine

Spearkiller committed:
Changed dogwood generation

Spearkiller committed:
Changed Macrocarpa generation

Spearkiller committed:
Update FeaturePadauk.java

Spearkiller committed:
Updated Camelthorn generation slightly

Spearkiller committed:
All trees now use the 'smart' branch algorithm

It doesn't seem to have broken anything

Spearkiller committed:
Updated acacia generation to match vanilla

Spearkiller committed:
Final touches to plum tree + padauk resprite

Spearkiller committed:
Reordered tree species, check desc.

Trees are ordered first by the line they belong to, and then by the tier they belong in. The tier is the number of 'steps' required to get back to a vanilla sapling. (E.G. oak is T0, Lime is T1, Cherry is T2, etc.)

Lines are sorted by:
Temperate
Swamp
Floral
Coniferous
Jungle
Arid

Spearkiller committed:
Overhauled mutations for Tree Breeding

Spearkiller committed:
Minor updates to sapling sprites

Spearkiller committed:
Update sapling.tree_sipiri.png

Spearkiller committed:
Added Amber, Amber Saplings and Amber Drones (unused)

These are currently unobtainable, and will require modifying sniffer loot which doesnt exist on this branch yet

Spearkiller committed:
Added squeezer uses for new fruits, rebalanced old recipes

Cherry now produces Fruit Juice instead of seed oil
Walnut now produces only 50mb of seed oil
Chestnut now produces only 80mb of seed oil

Coconut produces water for now, as milk is not available.

Spearkiller committed:
Update foresters_manual.png

Spearkiller committed:
Palm wood -> Date wood

Spearkiller committed:
Updated ginkgo generation to be truer to life

ExtraTrees showed me the light. One of the few trees they add that generates sensibly and realistically

Spearkiller committed:
A bunch of misc. resprites

Spearkiller committed:
Update sapling.tree_ebony.png

Spearkiller committed:
Updated chromosomes for new trees

No real rhyme or reason for them, but something different for each

Spearkiller committed:
Changed sappiness of trees

Also fixed a crash caused by me but also i blame autocomplete

Spearkiller committed:
Added Amber Thermionic Tubes

Spearkiller committed:
Fixed recipes from previous branches erroneously going missing

I hope I got them all

Spearkiller committed:
Amber Electron Tubes now apply a fortune effect

Only squeezers and centrifuges benefit from it though LOL

Spearkiller committed:
Amber, Amber Drones, and Amber Saplings can now be obtained legitimately

Spearkiller committed:
Rerun datagen to actually make the recipes :facepalm:

Spearkiller committed:
Resprited some fruits

Spearkiller committed:
Updated Teak Generation

Spearkiller committed:
updated textures for Humus and Coconut

Spearkiller committed:
Update stripped_poplar_log.png

Spearkiller committed:
Update orange.png

Spearkiller committed:
Mulch and Compost can be used as Bone Meal.

Using mulch, compost, or fertilizer now also makes a sound

Spearkiller committed:
Added descriptions to trees and JEI descriptions for mulch, compost, and fertilizer

Spearkiller committed:
Updated Ipe texture and breeding recipe

Spearkiller committed:
Updated a few tree textures

Spearkiller committed:
Updated Ginkgo log textures

Spearkiller committed:
Update stripped_ginkgo_log.png

Spearkiller committed:
Updated some analyser descriptions

Spearkiller committed:
resprited fruit leaves

Spearkiller committed:
slight change to macrocarpa door texture

Spearkiller committed:
Updated pods textures for pod fruits

Spearkiller committed:
Made it so coconuts spawn on coconut logs

It's a lot harder than it seems

Spearkiller committed:
Updated Papaya generation and sprites

Spearkiller committed:
Update sapling.tree_papaya.png

Spearkiller committed:
Update .gitignore

Spearkiller committed:
fixed a couple of things that broke during the merge

Spearkiller committed:
rerun datagen

Spearkiller committed:
Coconuts now produce milk instead of water

thedarkcolour commented on src/main/java/forestry/api/arboriculture/ForestryFruits.java:
New fruits:
- Pear
- Orange
- Coconut
- Feijoa
- Olive

thedarkcolour commented on src/main/java/forestry/api/arboriculture/ForestryTreeSpecies.java:
```diff
@@ -37,18 +37,32 @@ public class ForestryTreeSpecies {
 	public static final ResourceLocation BAOBAB = forestry("tree_baobab");
 	public static final ResourceLocation MAHOE = forestry("tree_mahoe");
 	public static final ResourceLocation WILLOW = forestry("tree_willow");
-	public static final ResourceLocation SIPIRI = forestry("tree_sipiri");
```
Did you mean to rename "Sipiri" to "Sipiri"?

Spearkiller replied:
Sipiri wasn't renamed in this version - in 1.21.1 there are some trees that need to be renamed internally.
Hill Cherry -> Sour Cherry
Citrus -> Lemon
Zebrawood -> Zebrano
Yellow Meranti -> Mahogany
Desert Acacia -> Camelthorn
and of course, Sipiri -> Greenheart

thedarkcolour commented on src/main/java/forestry/api/arboriculture/ForestryTreeSpecies.java:
New species:
- Elm
- Fir
- Coconut
- Beech
- Feijoa
- Dogwood
- Ginkgo
- Jacaranda
- Monkey Puzzle
- Macrocarpa
- Olive
- Orange
- Pear
- Kauri

Spearkiller committed:
Fixed Saturation values being WAY off

Spearkiller committed:
Changed Olive Mutation

Sour Cherry + Camelthorn to Sour Cherry + Wenge

Spearkiller committed:
Updated fruit values again

Spearkiller committed:
Made Papaya and Ginkgo drop fewer saplings

Spearkiller committed:
Fixed some trees not generating properly

thedarkcolour commented on src/main/java/forestry/factory/tiles/TileCentrifuge.java:
```diff
@@ -135,8 +135,8 @@ public boolean workCycle() {
 		// We are done, add products to queue
-		Collection<ItemStack> products = this.currentRecipe.getProducts(this.level.random);
-        this.pendingProducts.addAll(products);
+		Collection<ItemStack> products = currentRecipe.getProducts(level.random, outputMultiplier);
```
Qualify instance fields with `this.`

thedarkcolour commented:
<img width="835" height="758" alt="image" src="https://github.com/user-attachments/assets/92727a7a-a59b-46d4-a793-f26b263daa8b" />
Coconut hitbox issue

thedarkcolour committed:
Minor tweaks
- Rename Sipiri to Greenheart, Zebrawood to Zebrano, Monkey Puzzle to Pewen
- Formatting fixes for most of the tree features

thedarkcolour committed:
Fix typos using "it's" instead of "its"

thedarkcolour committed:
Add deprecated fallback method to ICentrifugeRecipe.getProducts

thedarkcolour committed:
Fix hitboxes for Coconut pods

thedarkcolour committed:
Fix uncaught renames

thedarkcolour committed:
Rerun datagen with new names + fix accidental Zebrawood ID change

thedarkcolour committed:
Fix crazy coconut generation + make pods respect Yield chromosome

thedarkcolour committed:
Sapling models are now data-generated instead of hand-written
